src/root/views/Country/Preparedness/EapApplication/useEapFormOptions.ts

```diff
@@ -0,0 +1,226 @@
+export type CountryDistrictsSchema = ArraySchema<PartialForm<CountryDistrictType>>;
+export type CountryDistrictsSchemaMember = ReturnType<CountryDistrictsSchema['member']>;
+
+export const schema: FormSchema = {
+  fields: (value): FormSchemaFields => ({
+    eap_number: [],
+    approval_date: [requiredCondition],
+    status: [requiredCondition],
+    operational_timeframe: [positiveIntegerCondition],
+    lead_time: [positiveIntegerCondition],
+    eap_timeframe: [positiveIntegerCondition],
+    num_of_people: [positiveIntegerCondition],
+    total_budget: [positiveIntegerCondition],
+    readiness_budget: [positiveIntegerCondition],
+    pre_positioning_budget: [positiveIntegerCondition],
+    early_action_budget: [positiveIntegerCondition],
+    trigger_statement: [requiredCondition],
+    overview: [requiredCondition],
+    document: [],
+    early_actions: [],
+    budget_per_sector: [],
+    prepositioning_activities: [],
+    originator_name: [],
+    originator_title: [],
+    originator_email: [],
+    originator_phone: [],
+    nsc_name: [],
+    nsc_title: [],
+    nsc_email: [],
+    nsc_phone: [],
+    ifrc_focal_name: [],
+    ifrc_focal_title: [],
+    ifrc_focal_email: [],
+    ifrc_focal_phone: [],
+    country: [requiredCondition],
+    district: [],
+    disaster_type: [],
+    indicator_value: [],
+    early_act: [],
+  })
+};
+
+const limitQuery = {
+  limit: 500,
+};
+
+interface EapOptions {
+  status: StringKeyValuePair[];
+  early_actions_indicators: StringKeyValuePair[];
+  sectors: StringKeyValuePair[];
+}
+
+function transformKeyValueToLabelValue<O extends NumericKeyValuePair | StringKeyValuePair>(o: O): {
+  label: string;
+  value: O['key'];
+} {
+  return {
+    value: o.key,
+    label: o.value,
+  };
+}
+
+function useEapsFormOptions(value: PartialForm<EapsFields>) {
+
+  const {
+    pending: fetchingEapOptions,
+    response: eapOptions,
+  } = useRequest<EapOptions>({
+    url: `api/v2/eap-options/`,
+  });
+
+  const [statusOptions] = React.useMemo(() => {
+    if (!eapOptions) {
+      return emptyStringOptionList;
+    }
+    return [
+      eapOptions.status.map(transformKeyValueToLabelValue),
+    ];
+  }, [eapOptions]);
+
+  const earlyActionIndicatorsOptions = React.useMemo(() => {
+    if (!eapOptions) {
+      return emptyStringOptionList;
+    }
+    return eapOptions.early_actions_indicators.map(transformKeyValueToLabelValue);
+  }, [eapOptions]);
+
+  const sectorsOptions = React.useMemo(() => {
+    if (!eapOptions) {
+      return emptyStringOptionList;
+    }
+    return eapOptions.sectors.map(transformKeyValueToLabelValue);
+  }, [eapOptions]);
+
+  const {
+    pending: fetchingEapDetails,
+    response: eapDetailsResponse,
+  } = useRequest<ListResponse<EapsFields>>({
+    url: `api/v2/eap/`,
+  });
+
+  const {
+    pending: fetchingDisasterTypes,
+    response: disasterTypesResponse,
+  } = useRequest<ListResponse<Disaster>>({
+    url: 'api/v2/disaster_type/',
+  });
```

Comment:
@frozenhelium , can we somehow call this separately so that we don't call it in each form?